tool/tctl/common/token_command.go

```diff
@@ -218,12 +218,23 @@ func (c *TokenCommand) Add(client auth.ClientI) error {
 			token,
 			int(c.ttl.Minutes()))
 	default:
+		proxies, err := client.GetProxies()
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		authServer := authServers[0].GetAddr()
+
+		if len(proxies) != 0 {
+			authServer = proxies[0].GetPublicAddr()
+		}
+
```

Comment:
Does this mean we'll always switch to displaying the proxy as how to join the cluster? Not sure we want to do that, joining the cluster via the auth server is a legitimate use case.

Reply:
I see, we don't wanna do that then.

Do you think we should add a new config variable and have the server signal to the client that it should join via proxy?
Another way I can think of is printing the URL the client actually used to authenticate.

To give you context, the problem is that the auth server is not always reachable to the node, especially on the cloud.

Please let me know your thoughts

Reply:
Sorry for commenting this late on the PR.

I think we might be able to work around this by looking at the Teleport license and default to returning the proxy address if we're running in Cloud.